### PR TITLE
Fix key signature error page

### PR DESCRIPTION
**Pull request by Gusted**
- When the GPG key contains an error, such as an invalid signature or an email address that does not match the user.A page will be shown that says you must provide a signature for the token.
- This page had two errors: one had the wrong translation key and the other tried to use an undefined variable [`.PaddedKeyID`](https://github.com/go-gitea/gitea/blob/e81ccc406bf723a5a58d685e7782f281736affd4/models/asymkey/gpg_key.go#L65-L72), which is a function implemented on the `GPGKey` struct, given that we don't have that, we use [`KeyID`](https://github.com/go-gitea/gitea/blob/e81ccc406bf723a5a58d685e7782f281736affd4/routers/web/user/setting/keys.go#L102) which is [the fingerprint of the publickey](https://pkg.go.dev/golang.org/x/crypto/openpgp/packet#PublicKey.KeyIdString) and is a valid way for opengpg to refer to a key.

Before:
image

After:
image

### CODE REVIEW

**Comment by Gusted** [comment]
Backport for v1.18: #22230
Backport for v1.17: #22231

File: templates/user/settings/keys_gpg.tmpl [hunk 1]
```diff
@@ -18,11 +18,11 @@
 					<p>{{.locale.Tr "settings.gpg_token_required"}}</p>
 				</div>
 				<div class="field">
-					<label for="token">{{.locale.Tr "setting.gpg_token"}}
+					<label for="token">{{.locale.Tr "settings.gpg_token"}}
 					<input readonly="" value="{{.TokenToSign}}">
 					<div class="help">
 						<p>{{.locale.Tr "settings.gpg_token_help"}}</p>
-						<p><code>{{$.locale.Tr "settings.gpg_token_code" .TokenToSign .PaddedKeyID}}</code></p>
+						<p><code>{{$.locale.Tr "settings.gpg_token_code" .TokenToSign .KeyID}}</code></p>
```
Comment (delvh) [inline templates/user/settings/keys_gpg.tmpl]: Wasn't that explicitly changed to only display the first 16 chars?
But apparently, this edge case was overlooked, so I'm fine with the proposed solution.

**Review by delvh — APPROVED** [review]

File: templates/user/settings/keys_gpg.tmpl (hunk 1 above)
Comment (Gusted) [reply templates/user/settings/keys_gpg.tmpl]: Looking at #20878, yeah that's indeed the case... If we want to have the same behavior, we would need to change https://github.com/go-gitea/gitea/blob/e81ccc406bf723a5a58d685e7782f281736affd4/models/asymkey/gpg_key_add.go#L96 to return the padded key ID. Or move the padding code to a util and handle it within the template.

CC @zeripath what's your thoughts?

Comment (zeripath) [reply templates/user/settings/keys_gpg.tmpl]: You're still gonna need the padding as keys that start with an initial 0 exist

**Review by zeripath — DISMISSED** [review]
See https://github.com/Gusted/gitea/pull/6

**Review by delvh — APPROVED** [review]

**Review by lunny — APPROVED** [review]

**Pull request merged**